internal/cli/atlas/backup/schedule/update.go

```diff
@@ -43,35 +56,163 @@ func (opts *UpdateOpts) initStore(ctx context.Context) func() error {
+	out := new(atlas.CloudProviderSnapshotBackupPolicy)
+
+	out.ClusterName = clusterName
+	opts.verifyExportBucketID(out)
+
+	if cmd.Flags().Changed(flag.ExportFrequencyType) {
+		checkForExport(out)
+		out.Export.FrequencyType = opts.exportFrequencyType
+	}
+	if cmd.Flags().Changed(flag.ReferenceHourOfDay) {
+		out.ReferenceHourOfDay = &opts.referenceHourOfDay
+	}
+	if cmd.Flags().Changed(flag.ReferenceMinuteOfHour) {
+		out.ReferenceMinuteOfHour = &opts.referenceMinuteOfHour
+	}
+	if cmd.Flags().Changed(flag.RestoreWindowDays) {
+		out.RestoreWindowDays = &opts.restoreWindowDays
+	}
+
+	out.AutoExportEnabled = returnValueForSetting(opts.autoExport, opts.noAutoExport)
+	out.UpdateSnapshots = returnValueForSetting(opts.updateSnapshots, opts.noUpdateSnapshots)
+	out.UseOrgAndGroupNamesInExportPrefix = returnValueForSetting(opts.useOrgAndGroupNamesInExportPrefix, opts.noUseOrgAndGroupNamesInExportPrefix)
+
+	return out, nil
+}
+
+func (opts *UpdateOpts) verifyExportBucketID(out *atlas.CloudProviderSnapshotBackupPolicy) {
+	if opts.exportBucketID != "" {
+		checkForExport(out)
+		out.Export.ExportBucketID = opts.exportBucketID
+	}
+}
+
+func (opts *UpdateOpts) verifyExportFrequencyType() func() error {
+	return func() error {
+		if opts.exportFrequencyType != "" {
+			if opts.exportFrequencyType != "daily" && opts.exportFrequencyType != "weekly" && opts.exportFrequencyType != "monthly" {
+				return errors.New("incorrect value for parameter exportFrequencyType. Value must be daily, weekly, or monthly")
+			}
+		}
+		return nil
+	}
+}
+
+func (opts *UpdateOpts) verifyReferenceHourOfDay(cmd *cobra.Command) func() error {
+	return func() error {
+		if cmd.Flags().Changed(flag.ReferenceHourOfDay) {
+			if opts.referenceHourOfDay < 0 || opts.referenceHourOfDay > 23 {
+				return errors.New("incorrect value for parameter referenceHourOfDay. Value must be an integer between 0 and 23 inclusive")
+			}
+		}
+		return nil
+	}
+}
+
+func (opts *UpdateOpts) verifyReferenceMinuteOfHour(cmd *cobra.Command) func() error {
+	return func() error {
+		if cmd.Flags().Changed(flag.ReferenceMinuteOfHour) {
+			if opts.referenceMinuteOfHour < 0 || opts.referenceMinuteOfHour > 59 {
+				return errors.New("incorrect value for parameter referenceMinuteOfHour. Value must be an integer between 0 and 59 inclusive")
+			}
+		}
+		return nil
+	}
+}
+
+func (opts *UpdateOpts) verifyRestoreWindowDays(cmd *cobra.Command) func() error {
+	return func() error {
+		if cmd.Flags().Changed(flag.RestoreWindowDays) {
+			if opts.restoreWindowDays <= 0 {
+				return errors.New("incorrect value for parameter restoreWindowDays. Value must be a positive, non-zero integer")
+			}
+		}
+		return nil
+	}
+}
+
+func checkForExport(out *atlas.CloudProviderSnapshotBackupPolicy) {
+	if out.Export == nil {
+		out.Export = new(atlas.Export)
+	}
+}
+
+func returnValueForSetting(enableFlag, disableFlag bool) *bool {
```

Comment:
I think I've seen it before no?
can we reuse it across packages?

Reply:
Sure! I moved it to another package.